scenario/mocking.py

```diff
@@ -632,11 +632,16 @@ def resource_get(self, resource_name: str) -> str:
             )
 
     def credential_get(self) -> CloudSpec:
-        if not self._state.cloud_spec:
+        if not self._state.trusted:
```

Comment:
have you thought about putting trusted in `Context` instead of `State`?
it feels like it crosses concern boundaries here. Philosophically, State = hook tool mocks. This is more meta: it represents whether the charm was deployed with `--trust` or not. 
Perhaps we should put it on the same level as juju_version. Not quite state, but still a runtime concern --> goes in Context.

Perhaps I'm overthinking it?

Reply:
Hmm, good point. I was thinking more of the "run juju trust x" case than deploying with trust, so thinking in the context of running a CLI command that caused a change, which seemed State-ish. But indeed, there's no hook tool to get trust (and obviously couldn't be one to set it)

I'm convinced; will adjust.